### PR TITLE
 Add logic for schedule_date 

### PR DESCRIPTION
**Pull request by clarud**
Implement logic for ScheduleDateCommand

Previously command had no logic.
Now it can display the schedule on a specified date.
Fix minor parsing in ScheduleDateParser.
Add test for ScheduleDateCommand

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/ScheduleDateCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        String expectedMessage = String.format(MESSAGE_APPOINTMENTS_LISTED_OVERVIEW, 0, "29 February 2024");
+        AppointmentContainsDatePredicate predicate = preparePredicate("29-02-2024");
+        ScheduleDateCommand command = new ScheduleDateCommand(predicate);
+        expectedModel.updateFilteredAppointmentList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredAppointmentList());
+    }
+
+    @Test
+    public void execute_multipleKeywords_multiplePersonsFound() {
```
Comment (choonzies) [inline src/test/java/seedu/address/logic/commands/ScheduleDateCommandTest.java:68]: Just want to clarify if multipleKeywords corresponds to the case where there multiple people with appointments on the date queried?

**Review by choonzies — COMMENTED** [review]
Implementation LGTM.

File: src/test/java/seedu/address/logic/commands/ScheduleDateCommandTest.java (hunk 1 above)
Comment (clarud) [reply src/test/java/seedu/address/logic/commands/ScheduleDateCommandTest.java:68]: si

File: src/main/java/seedu/address/logic/commands/ScheduleDateCommand.java [hunk 2]
```diff
@@ -1,35 +1,43 @@
...
+        requireNonNull(predicate);
+        this.predicate = predicate;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
-        // TODO: Implement logic to filter and display appointments for the given date
-        return new CommandResult(String.format(MESSAGE_SUCCESS, date));
+        requireNonNull(model);
+        model.updateFilteredAppointmentList(predicate);
```
Comment (niclammm) [inline src/main/java/seedu/address/logic/commands/ScheduleDateCommand.java:36]: Good use of existing person predicate class. Good job

File: src/test/java/seedu/address/logic/commands/ScheduleDateCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+ * Contains integration tests (interaction with the Model) for {@code ScheduleDateCommand}.
+ */
+public class ScheduleDateCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        AppointmentContainsDatePredicate firstPredicate = preparePredicate("24-10-2024");
```
Comment (niclammm) [inline src/test/java/seedu/address/logic/commands/ScheduleDateCommandTest.java:34]: I think we could add these declarations of predicates into a new class under test util to allow for better readability. But other than that, straight forward and clear testing good work

**Review by niclammm — COMMENTED** [review]
Overall logic is correct and minor changes needed

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 4]
```diff
@@ -20,10 +22,14 @@ public class TypicalPersons {
...
             .withId("P12345").withWard("A1").withDiagnosis("Celiac Disease")
-            .withMedication("gluten-free diet").withNotes("prone to falling").build();
+            .withMedication("gluten-free diet").build();
     public static final Person BENSON = new PersonBuilder().withName("Benson Meier")
             .withId("P54321").withWard("A2").withDiagnosis("Celiac Disease")
-            .withMedication("gluten-free diet").build();
+            .withMedication("gluten-free diet").withNotes("some notes")
+            .withAppointment("surgery",
+                    LocalDateTime.of(2024, 11, 11, 11, 0),
+                    LocalDateTime.of(2024, 11, 11, 11, 50))
```
Comment (niclammm) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:31]: Correct use and edit of typical persons but resulted in conflict with other prs

**Pull request merged**